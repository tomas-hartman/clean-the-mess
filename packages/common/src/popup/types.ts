import { Tabs } from 'webextension-polyfill';
import { EnumGuard, ValueOf } from '../../types';

export type OverviewItem = {
  url: Tabs.Tab['url'];
  count: number;
  ids: number[];
  key: string;
  favicon?: Tabs.Tab['favIconUrl'];
};

export type ScreensProps = {
  details: {
    ids: OverviewItem['ids'];
    url?: string;
    key: string;
    hasActionButton?: boolean;
  };
  latest: undefined;
  search: undefined;
  overview: undefined;
  duplicates: undefined;
};

export const SCREEN = {
  OVERVIEW: 'overview',
  DETAILS: 'details',
  LATEST: 'latest',
  SEARCH: 'search',
  DUPLICATES: 'duplicates',
} as const satisfies EnumGuard<ScreensProps>;

export type ScreenName = ValueOf<typeof SCREEN>;

export type ScreenProps = {
  name: ScreenName;
  options?: ScreensProps[ScreenName];
};
