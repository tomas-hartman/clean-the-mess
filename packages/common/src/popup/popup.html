<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  
<!-- <link rel="stylesheet" href="../styles/chrome/style.scss"> -->

  <title>Clean the mess! addon</title>
</head>

<body>
  <main id="main-container"></main>
  <div id="modal-root"></div>
  <script type="module" src="./Popup.tsx"></script>
</body>

</html>