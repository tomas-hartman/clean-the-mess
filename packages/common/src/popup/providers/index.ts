export * from "./NavigationProvider"
export * from "./DataProvider"