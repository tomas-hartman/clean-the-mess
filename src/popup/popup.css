:root {
  --color-link-hover-purple: #c169ffb9;
  --color-link-hover: var(--color-emphasis-dark);
  --color-emphasis-dark: #d7d7dbdc;
  --color-emphasis-light: #ededf0d0;
  --color-separator: #d7d7db;
  --color-background: #ffffff;
  --color-font: #0c0c0d;
  --color-secondary-font: #737373;
  --color-back-button: var(--color-background);
}

@media (prefers-color-scheme: dark) {
  :root {
    --color-link-hover: #663399;
    --color-emphasis-dark: #737373c1;
    --color-emphasis-light: #7373738a;
    --color-separator: #7373738a;
    --color-background: #4a4a4f;
    --color-font: #ffffff;
    --color-secondary-font: #f9f9fa;
    --color-back-button: darkorchid;
  }
}

html,
body {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  padding: 10px;
  margin: 0;
  padding: 0;
  max-width: 370px;
  min-width: 350px;
  width: auto;
  overflow: hidden;
  background-color: var(--color-background);
  color: var(--color-font);
  
  font-size: 12px;
  font-weight: 400;
  font-family: 'Segoe UI', 'San Francisco', 'Helvetica Neue', 'Ubuntu', 'Roboto', 'sans-serif';
  font: caption;
  
  cursor: default;
  user-select: none;
  -moz-user-select: none;
}

.hidden {
  display: none;
}

ul {
  list-style: none;
  margin-block-start: auto;
  margin-block-end: auto;
  padding-inline-start: initial;
}

li {
  display: flex;
  width: 100%;
  height: 20px;
  padding: 4px 0;
  align-items: center;
}

li:hover,
#ten-unused:hover {
  background-color: var(--color-link-hover);
}

li>div {
  display: flex;
  width: 100%;
  box-sizing: border-box;
  padding: 0 12px 0 16px;
  align-items: center;
  flex-basis: auto;
  align-items: flex-start;
  align-items: center;
  justify-content: space-between;
}

.url-container>div,
.main-item-text-container>div,
.item-container>div,
.item-buttons-container>div {
  margin-left: 4px;
}

.url-container>div:first-of-type,
.main-item-text-container>div:first-of-type,
.item-container>div:first-of-type,
.item-buttons-container>div:first-of-type {
  margin-left: 0;
}

.item-text-container {
  min-width: 0;
}

.item-buttons-container {
  flex-shrink: 0;
  display: flex;
  min-width: 0;
}

.main-item-text-container {
  display: flex;
  width: inherit;
  justify-content: space-between;
  min-width: 0;
}

#header>span,
#ten-unused>span,
.header-title > span {
  padding: 4px 16px;
  display: inline-block;
  height: 20px;
  line-height: 20px;
}

#ten-unused>span {
  padding: initial;
  height: initial;
}

#ten-unused {
  margin-top: 4px;
  display: inline-grid;
  grid-template-columns: [text] auto 20px [end];
  width: 100%;
  padding: 4px 16px;
  box-sizing: border-box;
  align-items: center;
}

.separator {
  border-bottom: 1px solid var(--color-separator);
  margin-bottom: 4px;
  margin-top: 4px;
}

.separator-bottom {
  margin-bottom: 0;
}

#main-container {
  left: 0px;
  position: relative;
  min-height: 300px;
  max-height: 600px;
  display: flex;
}

.body-container, .ul-container {
  overflow-y: scroll;
  overflow-y: auto;
}

.body-container {
  display: flex;
  width: 100%;
  flex-direction: column;
}

.header-container {
  margin-top: 4px;
}

.url,
.last-displayed {
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.count {
  flex-basis: auto;
}

.remove,
.get-in,
.bookmark {
  width: 20px;
  height: 20px;
  align-self: center;
  text-align: center;
  flex-shrink: 0;
}

.remove {
  background-image: url(../icons/btn-remove.svg);
  background-size: 75%;
  background-repeat: no-repeat;
  background-position: center;
  margin-left: 4px !important;
}

.remove:hover,
.bookmark:hover {
  background-color: var(--color-emphasis-light);
}

.remove:active,
.bookmark:active {
  background-color: var(--color-emphasis-dark);
}

.bookmark {
  border-radius: 1px;
  background-image: url(../icons/btn-bookmark-star.svg);
  background-size: 70%;
  background-repeat: no-repeat;
  background-position: center;
}

.bookmark.bookmark-close {
  background-image: url(../icons/ico-bookmark-close.svg);
  background-size: 70%;
}

.bookmark.bookmarked {
  background-image: url(../icons/btn-bookmark-star-full.svg);
}

.get-in {
  background-image: url(../icons/btn-bold-arrow.svg);
  background-size: 60%;
  background-repeat: no-repeat;
  background-position: center;
}

.go-back {
  background-image: url(../icons/btn-big-arrow.svg);
  background-size: 50%;
  background-repeat: no-repeat;
  background-position: center;
  transform: rotate(180deg);
}

/* DETAILS */

#overview {
  width: 100%;
  display: flex;
  flex-direction: column;
}

#details, #latest {
  position: absolute;
  height: 100%;
  width: 100%;
  right: -100%;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
}

#header.control {
  display: grid;
  grid-template-columns: 40px auto 40px;
  align-content: center;
  height: 30px;
}

.control .back, .control .close-all {
  justify-self: center;
  width: 32px;
  height: 32px;
  background-color: var(--color-back-button);
  align-self: center;
  text-align: center;
  font-size: 1.15em;
  border-radius: 1px;
  display: block;
  line-height: 32px;
}

.control .back:hover {
  background-color: var(--color-emphasis-light);
}

.control .back:active {
  background-color: var(--color-emphasis-dark);
}

.control .close-all {
  background-color: inherit;
  background-image: url("../icons/btn-big-remove.svg");
  background-size: 60%;
  background-position: center;
  background-repeat: no-repeat;
}

.control .close-all:hover {
  background-color: var(--color-emphasis-light);
}

.control .close-all:active {
  background-color: var(--color-emphasis-dark);
}

.header-title {
  justify-self: center;
  align-self: center;
}

.title.detail {
  overflow-x: hidden;
  text-overflow: ellipsis;
  word-break: keep-all;
  white-space: nowrap;
}

.url.detail,
.last-displayed.detail {
  font-style: italic;
  word-break: keep-all;
}

li.detail {
  height: 40px;
}

/* ANIMATIONS */
.slide-in {
  right: 0% !important;

  transition: right 150ms cubic-bezier(.07, .95, 0, 1);
}

.slide-out {
  left: -100% !important;

  transition: left 150ms cubic-bezier(.07, .95, 0, 1);
  position: relative;
}

#overview.slide-out {
  overflow-y: hidden !important;
}

.slide-out-reverse {
  right: -100% !important;

  transition: right 150ms cubic-bezier(.07, .95, 0, 1);
}

.slide-in-reverse {
  left: 0% !important;

  transition: left 150ms cubic-bezier(.07, .95, 0, 1);
}

#overview.slide-in-reverse {
  overflow-y: auto !important;
}

.screen {
  top: 0;
}