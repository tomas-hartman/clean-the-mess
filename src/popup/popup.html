<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <link rel="stylesheet" href="popup.css">
  </head>
  <body>
    <main id="main-container">
      <div id="header"></div>
      <div class="separator separator-bottom"></div>
      <div id="main"></div>
    </main>
    <script src="popup.js"></script>
  </body>
</html>