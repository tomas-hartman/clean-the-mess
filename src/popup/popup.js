(async () => {
    const locale = {
        string: "cs-CZ",
        options: {
            year: 'numeric', month: 'numeric', day: 'numeric',
            hour: 'numeric', minute: 'numeric', second: 'numeric' 
        }
    }

    let tabs = await browser.tabs.query({currentWindow: true});
    console.log(tabs);
    const windows = await browser.windows.getAll();
    let tabsOverview = []; // fills with getOverview
    const anchorMain = document.querySelector("#main");
    const anchorHeader = document.querySelector("#header");
    const windowStr = windows.length > 1 ? " in this window" : "";
    const openedTabsStr = `<span>You have <span id="open-tabs-count">${tabs.length}</span> opened tabs${windowStr}.</span>`;

    anchorHeader.appendChild(document.createRange().createContextualFragment(openedTabsStr));

    /**
     * @todo Work on detailed and better filtered return array
     * @param {array} tabs tabs query array 
     * @param {number} numOfLatest optional, is equal to 10 normally 
     */
    const getLatestUsed = (tabs, numOfLatest = 10) => {
        let newTabs = tabs.slice(0);
        let iterationsNum = numOfLatest;
        let latest = [];

        if(tabs.length < iterationsNum) iterationsNum = tabs.length; 

        newTabs.sort((a,b) => a.lastAccessed - b.lastAccessed);

        for(let i=0;i<iterationsNum;i++){
            if(newTabs[i].pinned) continue;

            let output = {};
            let date = new Intl.DateTimeFormat(locale.string, locale.options).format(new Date(newTabs[i].lastAccessed));
            
            output.date = date;
            output.title = newTabs[i].title;
            output.id = newTabs[i].id;
            output.url = newTabs[i].url;

            latest.push(output);
        }

        return latest;
    }

    const getOverview = (tabs) => {
        const tabsOverview = [];

        tabs.forEach(tab => {
            let url = new URL(tab.url);
            let originUrl = url.origin;

            if(originUrl === "null" || url.protocol === "moz-extension:"){
                switch (url.protocol) {
                    case "about:":
                    case "moz-extension:":
                        originUrl = "Browser tabs";
                        break;
                    case "file:":
                        originUrl = "Opened files";
                        break;
                    default:
                        originUrl = "Other tabs";
                        break;
                }
            };
            
            if(!tab.pinned){
                if (tabsOverview.some(website => website.url === originUrl)) {
                    
                    const index = tabsOverview.findIndex( website => website.url === originUrl );

                    tabsOverview[index].count += 1;
                    tabsOverview[index].ids.push(tab.id);
    
                } else {
                    tabsOverview.push({ url: originUrl, count: 1, ids: [ tab.id ] });
                }
            }
        });
        
        tabsOverview.sort((a,b) => b.count-a.count);

        return tabsOverview;
    }
    tabsOverview = getOverview(tabs);

    const getOverviewList = (tabsOverview) => {
        const ul = document.createElement("ul");
              ul.setAttribute("id", "list");
    
        for(let i = 0; i < tabsOverview.length; i++){
            let tab = tabsOverview[i];

            let text = `<li class="url-${i}" data-index-number="${i}">
                            <div class="url-container">
                                <div class="main-item-text-container">
                                    <div class="url" title="${tab.url}">${tab.url}</div>
                                    <div class="count">(${tab.count})</div>
                                </div>
                                <div class="item-buttons-container">
                                    <div class="get-in"></div>
                                    <div class="remove hidden" title="Close all tabs with this url"></div>
                                </div>
                            </div>
                        </li>`;
            
            ul.appendChild(document.createRange().createContextualFragment(text));
        }

        return ul;
    }

    anchorMain.appendChild(getOverviewList(tabsOverview));

    /**
     * Use only after removal!
     * @param {string} data id of selected url from tabsOverview
     */
    const refreshOverviewData = (data) => {
        let currentTabsNum = parseInt(document.querySelector("#open-tabs-count").innerText);
        let tabsNum = tabsOverview[data].ids.length;
        let newTabsNum = currentTabsNum - tabsNum;
        const openedTabsStr = `<span>You have <span id="open-tabs-count">${newTabsNum}</span> opened tabs${windowStr}.</span>`

        document.querySelector(`li.url-${data}`).remove();

        anchorHeader.firstChild.remove();
        anchorHeader.appendChild(document.createRange().createContextualFragment(openedTabsStr));
    }

    /**
     * Use only after details to overview transition (pressing back button)
     */
    const refreshOverviewScreen = async () => {
        document.querySelector("#main-container").classList.add("slide-in-reverse");
        document.querySelector("#details").classList.add("slide-out-reverse");
        
        document.addEventListener("transitionend", () => {
            document.querySelector("#main-container").classList = "";
        }, { once: true });
        
        tabs = await browser.tabs.query({currentWindow: true});
        tabsOverview = getOverview(tabs);

        const openedTabsStr = `<span>You have <span id="open-tabs-count">${tabs.length}</span> opened tabs${windowStr}.</span>`

        anchorHeader.firstChild.remove();
        anchorHeader.appendChild(document.createRange().createContextualFragment(openedTabsStr));

        anchorMain.removeChild(anchorMain.firstChild);
        anchorMain.appendChild(getOverviewList(tabsOverview));
    }

    const removeTabs = (target) => {
        const data = target.closest("li").dataset.indexNumber;
        const id = tabsOverview[data].ids;
        if(id.length > 10) {
            if(confirm(`Are you sure you want to close ${id.length} tabs?`)){
                browser.tabs.remove(id);
                refreshOverviewData(data);
            } else return;
        } else {
            browser.tabs.remove(id);
            refreshOverviewData(data);
        }
    }

    const getDetails = (target) => {
        const index = target.closest("li").dataset.indexNumber;
        const ids = tabsOverview[index].ids;
        let array = [];

        for(let i=0;i<ids.length;i++){
            array.push(... tabs.filter(tab => tab.id === ids[i]));
        }

        array.sort((a, b) => b.lastAccessed-a.lastAccessed)

        return array;
    }

    const addBookmarkStatus = async (item) => {
        const bookmarks = await browser.bookmarks.search({url: item.url});
        const elm = document.querySelector(`li[data-tab-id='${item.id}'] .bookmark`);

        if(bookmarks.length > 0){
            // TBA After design is ready
            elm.classList.add("bookmarked");
            elm.setAttribute("title", "This url is already bookmarked");
        }
    }

    const createSlideScreen = (headerTitle) => {
        const headerDivStr = `<div id="header" class="control"><div class="back go-back" title="Back"></div>
                        <div class="header-title">${headerTitle}</div></div>
                        <div class="separator separator-bottom"></div>`;
        const headerDiv = document.createRange().createContextualFragment(headerDivStr);

        const mainDetailsDiv = document.createElement("main");
              mainDetailsDiv.setAttribute("id", "details");
              mainDetailsDiv.appendChild(headerDiv);

        const ul = document.createElement("ul");
        const ulContainer = document.createElement("div");
              ulContainer.classList.add("ul-container");

        ulContainer.appendChild(ul);
        mainDetailsDiv.appendChild(ulContainer);

        return mainDetailsDiv;
    }

    /**
     * @todo Merge showLatestDisplayed() with showDetailsScreen()
     * @param {DOM target} target 
     */
    const showLatestDisplayed = (target, numOfLatest = 10) => {
        const array = getLatestUsed(tabs, numOfLatest);
        let headerTitle = `${numOfLatest} longest unused tabs`;

        const screen = createSlideScreen(headerTitle);
        const ul = screen.querySelector("ul");

        // Fill in with content
        for (let i = 0; i < array.length; i++) {
            const text = `
                <li id="item-${i}" class="detail" data-tab-id="${array[i].id}">
                    <div class="item-container detail">
                        <div class="item-text-container">
                            <div class="title detail" title="${array[i].title}">${array[i].title}</div>
                            <div class="url detail hidden" title="${array[i].url}">${array[i].url}</div>
                            <div class="last-displayed detail" title="${array[i].date}">${array[i].date}</div>
                        </div>
                        <div class="item-buttons-container">
                            <div class="bookmark bookmark-close detail" title="Bookmark and close tab"></div>
                            <div class="remove detail" title="Close tab"></div>
                        </div>
                    </div>
                </li>
                `;
            
        ul.appendChild(document.createRange().createContextualFragment(text));
        addBookmarkStatus(array[i]);
        }

        // Set up events
        screen.onclick = (e) => {
            if(e.target.classList.contains("back")){

                refreshOverviewScreen();

                document.querySelector("#details").remove();
                document.querySelector("#main-container").style.left = "0px";
            }
            if(e.target.classList.contains("remove")){
                // remove tab
                const id = parseInt(e.target.closest("li").dataset.tabId);
                browser.tabs.remove([id]);
                e.target.closest("li").remove();
            }
            if(e.target.closest("li") && !e.target.classList.contains("remove")){
                // switchTo tab
                const id = parseInt(e.target.closest("li").dataset.tabId);
                browser.tabs.update(id, {active: true});
            }
        }

        document.querySelector("body").appendChild(screen);
    }

    const showDetailsScreen = async (target) => {
        let array = getDetails(target);
        let headerTitle = "";

        try {
            headerTitle = (new URL(target.querySelector('.url').innerText)).host;
        } catch (error) {
            headerTitle = target.querySelector('.url').innerText;
        }
        
        const screen = createSlideScreen(headerTitle);
        const ul = screen.querySelector("ul");

        // Fill in with content
        for(let i=0; i < array.length; i++){
            const text = `
            <li id="item-${i}" class="detail" data-tab-id="${array[i].id}">
                <div class="item-container detail">
                    <div class="item-text-container">
                        <div class="title detail" title="${array[i].title}">${array[i].title}</div>
                        <div class="url detail" title="${array[i].url}">${array[i].url}</div>
                    </div>
                    <div class="item-buttons-container">
                        <div class="bookmark bookmark-close detail hidden" title="Bookmark and close tab"></div>
                        <div class="remove detail hidden" title="Close tab"></div>
                        <div class="get-in"></div>
                    </div>
                </div>
            </li>
            `;

            ul.appendChild(document.createRange().createContextualFragment(text));
            addBookmarkStatus(array[i]);
        }

        // Set up events
        screen.onmouseover = (e) => {
            if(e.target.closest("li")){
    
                const parentElm = e.target.closest("li").querySelector(".item-buttons-container");
                parentElm.children[0].classList.remove("hidden");
                parentElm.children[1].classList.remove("hidden");
                parentElm.children[2].classList.add("hidden");
            }
        }
        screen.onmouseout = (e) => {
            if(e.target.closest("li")){
    
                const parentElm = e.target.closest("li").querySelector(".item-buttons-container");
                parentElm.children[0].classList.add("hidden");
                parentElm.children[1].classList.add("hidden");
                parentElm.children[2].classList.remove("hidden");
            }
        }

        screen.onclick = (e) => {
            if(e.target.classList.contains("back")){
                refreshOverviewScreen();

                document.querySelector("#details").remove();
                document.querySelector("#main-container").style.left = "0px";
            }
            if(e.target.classList.contains("remove")){
                // remove tab
                const id = parseInt(e.target.closest("li").dataset.tabId);
                
                array.splice(array.findIndex(tab => tab.id === id), 1);
                browser.tabs.remove([id]);
                e.target.closest("li").remove();

                // autoclose
                if(array.length === 0){
                    refreshOverviewScreen();

                    document.querySelector("#details").remove();
                    document.querySelector("#main-container").style.left = "0px";
                }
            }
            if(e.target.closest("li") && !e.target.classList.contains("remove")){
                // switchTo tab
                const id = parseInt(e.target.closest("li").dataset.tabId);
                browser.tabs.update(id, {active: true});
            }
        }
        
        document.querySelector("body").appendChild(screen);
    }

    document.querySelector("#main-container").onmouseover = (e) => {
        if(e.target.closest("li")){

            const parentElm = e.target.closest("li").querySelector(".item-buttons-container");
            parentElm.children[1].classList.remove("hidden");
        }
    }
    document.querySelector("#main-container").onmouseout = (e) => {
        if(e.target.closest("li")){

            const parentElm = e.target.closest("li").querySelector(".item-buttons-container");
            parentElm.children[0].classList.remove("hidden");
            parentElm.children[1].classList.add("hidden");
        }
    }
    document.querySelector("#main-container").onclick = (e) => {
        if(e.target.classList.contains("remove")){
            removeTabs(e.target);
            console.log("Tabs removed!");
        }
        if( e.target.closest("div.url-container") && !e.target.classList.contains("remove")){
            // getDetails(e.target);
            const target = e.target.closest("div.url-container");

            showDetailsScreen(target);
            new Promise((resolve, reject) => {
                setTimeout(() => {
                    resolve();
                }, 10);
            }).then(() => {
                document.querySelector("#main-container").classList.add("slide-out");
                document.querySelector("#details").classList.add("slide-in");
            });

            console.log("clicked");
        }
        if(e.target.closest("#ten-unused")){
            showLatestDisplayed(e.target);

            new Promise((resolve, reject) => {
                setTimeout(() => {
                    resolve();
                }, 10);
            }).then(() => {
                document.querySelector("#main-container").classList.add("slide-out");
                document.querySelector("#details").classList.add("slide-in");
            });
        }
    }

    browser.theme.getCurrent().then((a) => {
        console.log(a);
    });

    getLatestUsed(tabs);

})();
