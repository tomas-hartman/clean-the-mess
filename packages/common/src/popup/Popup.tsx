import { DataProvider } from './providers/DataProvider';
import { NavigationProvider } from './providers';
import { Router } from './screens/Router';
import { useColorScheme } from './hooks';
import { createRoot } from 'react-dom/client';
import { isChrome } from './utils';

// Global theme style
import '../styles/global.css';

if (isChrome()) {
  import('../styles/themesChrome.css');
} else {
  import('../styles/themesFirefox.css');
}

export default function Popup() {
  useColorScheme();

  return (
    <DataProvider>
      <NavigationProvider>
        <Router />
      </NavigationProvider>
    </DataProvider>
  );
}

const container = document.getElementById('main-container');
const root = createRoot(container!); // createRoot(container!) if you use TypeScript
root.render(<Popup />);
