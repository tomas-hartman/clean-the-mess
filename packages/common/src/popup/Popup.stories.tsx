import type { Meta, StoryObj } from '@storybook/react';

import Popup from './Popup';

const meta: Meta<typeof Popup> = {
  title: 'Popup',
  component: Popup,
  decorators: [Story => <Story />],
};

export default meta;

type Story = StoryObj<typeof Popup>;

export const PopupComponent: Story = {
  args: {},
  name: 'Popup',
};
