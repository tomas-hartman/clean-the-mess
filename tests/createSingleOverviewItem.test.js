import { createSingleOverviewItem } from '../src/popup/popup';

it('should return correct component', () => {
  const data = {
    url: 'https://jestjs.io',
    count: 3,
    ids: [
      81,
      93,
      96,
    ],
  };

  const props = {
    itemId: 1,
    data,
  };

  const output = createSingleOverviewItem(props);

  expect(output).toMatchSnapshot();
});
