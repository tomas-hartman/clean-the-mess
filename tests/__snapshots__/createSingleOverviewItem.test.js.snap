// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct component 1`] = `
<DocumentFragment>
  
        
  <li
    class="url-1 overview-item"
    data-index-number="1"
  >
    
            
    <div
      class="url-container"
    >
      
                
      <div
        class="main-item-text-container"
      >
        
                    
        <div
          class="url"
          title="https://jestjs.io"
        >
          https://jestjs.io
        </div>
        
                    
        <div
          class="count"
        >
          (3)
        </div>
        
                
      </div>
      
                
      <div
        class="item-buttons-container"
      >
        
                    
        <div
          class="get-in"
        />
        
                    
        <div
          class="bookmark-all hidden"
          title="Bookmark and close all items"
        >
          BA
        </div>
        
                    
        <div
          class="remove hidden"
          title="Close all tabs with this url"
        />
        
                
      </div>
      
            
    </div>
    
        
  </li>
</DocumentFragment>
`;
